src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +38,14 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToObj(index -> !pairingCheck(index) ? numbers.get(index) - SUBTRACTION_VALUE
+                        : numbers.get(index))
+                .mapToInt(number -> number)
+                .filter(number -> !pairingCheck(number))
+                .average()
+                .orElseThrow(() -> new NoSuchElementException(
+                        "The average of odd numbers cannot be calculated"));
     }
```

Comment:
1. Boolean method are more understandable when named with 'is'. Like isEvenNumber, isValidRecord etc.
2. Ternaries are not easiest structure for reading so should be properly formatted
3. exclamation mark (!) in code is very easy to be missed. When not a problem it's better to create opposite method not to use !
4. Why mapToObj to return to int?
```suggestion
                .map(index -> isOddNumber(index) 
                        ? numbers.get(index) - SUBTRACTION_VALUE
                        : numbers.get(index))
                .filter(number -> isOddNumber(number))
                .average()
                .orElseThrow(() -> new NoSuchElementException(
                        "The average of odd numbers cannot be calculated"));
    }
```